src/Pact/Analyze/Translate.hs

```diff
@@ -112,15 +113,20 @@ describeTranslateFailureNoLoc = \case
 data TranslateEnv
   = TranslateEnv
     { _teInfo           :: Info
-    , _teNodeVars       :: Map Node (Text, VarId)
+    , _teNodeVars       :: Map Node (Munged, VarId)
     , _teRecoverability :: Recoverability
+    , _teScopesEntered  :: Natural
     }
 
 mkTranslateEnv :: Info -> [Arg] -> TranslateEnv
-mkTranslateEnv info args = TranslateEnv info nodeVars mempty
+mkTranslateEnv info args = TranslateEnv info nodeVars mempty 0
   where
+    -- NOTE: like in Check's moduleFunChecks, this assumes that toplevel
```

Comment:
Maybe we should introduce a function with a long name here, like `coerceArgumentToMunged`

Reply:
Good call!